### DISCUSSION TITLE
How to validate array in Goravel?

### DISCUSSION THREAD

**Discussion by corven13** [goravel/goravel#143] [Q&A]
I'm trying to validate an array of data. but can't

Input from Postman:
`
{
    "data": [

          {
            "id": 1440,
            "main_product": "AAAA",
            "product": "AA",
            "unit": "MMscfd",
            "amount": 10
        },
        {
            "id": 1441,
            "main_product": "BBBB",
            "product": "BB",
            "unit": "bbls/d",
            "amount": 20
        }
    ]
}

`

**Comment by hwbrzzl** [comment]
Hi @corven13 You need to implenment a custom validate rule to support this data structure, but we support array/slice now: ["a", "b"], by rule: `"names.*": "required|min_len:2"`

**Comment by istiyakamin** [comment]
To validate an array of objects in Goravel—a Laravel-inspired framework for Go—you can apply validation rules to each element within the array using dot notation and the * wildcard. This approach is similar to Laravel's validation mechanism.

`{
  "data": [
    {
      "id": 1440,
      "main_product": "AAAA",
      "product": "AA",
      "unit": "MMscfd",
      "amount": 10
    },
    {
      "id": 1441,
      "main_product": "BBBB",
      "product": "BB",
      "unit": "bbls/d",
      "amount": 20
    }
  ]
}
`

Validation Rules
To validate each object within the data array, you can define rules as follows:

`rules := map[string]string{
  "data":           "required|array",
  "data.*.id":      "required|integer",
  "data.*.main_product": "required|string",
  "data.*.product":      "required|string",
  "data.*.unit":         "required|string",
  "data.*.amount":       "required|numeric|min:0",
}
`
